arbos/programs/programs.go

```diff
@@ -256,19 +257,46 @@ func (p Programs) CallProgram(
 		// Ensure that return data costs as least as much as it would in the EVM.
 		evmCost := evmMemoryCost(uint64(len(ret)))
 		if startingGas < evmCost {
+			attributeWasmComputation(contract, startingGas)
```

Comment:
I believe used gasGas should be `startingGas - contract.Gas` here as well. In that case, attributeWasmComputation can receive startingGas and it can compute usedGas inside the function.

Reply:
fixed